core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -319,8 +313,9 @@ public QueryCachingPolicy getQueryCachingPolicy() {
      * unless explicitly disabled.
      *
      * @throws IndexShardRelocatedException if shard is marked as relocated and relocation aborted
+     * @throws IOException if shard state could not be persisted
      */
-    public void updateRoutingEntry(final ShardRouting newRouting, final boolean persistState) {
+    public void updateRoutingEntry(final ShardRouting newRouting, final boolean persistState) throws IOException {
```

Comment:
OK. So it's not strictly needed, just an improvement. wanted to make sure I didn't miss anything.